src/BuiltInTools/dotnet-watch/Program.cs

```diff
@@ -198,6 +198,7 @@ private async Task<int> RunAsync(CommandLineOptions options, CancellationToken c
 
             var context = new DotNetWatchContext
             {
+                LaunchProfileName = launchProfileName,
```

Comment:
Doesn't seem correct. If neither `--launch-profile` or `--no-launch-profile` are specified, the default (first that has Project command) profile from `launchSettings.json` is selected (if any). In that case `launchProfileName` will be null here but you'll want to set the env variable to the default profile name, no?

Reply:
Right, this should be something like `LaunchProfileName = launchProfile.Name` instead? Which would require adding `Name` to `LaunchSettingsProfile`.

Reply:
Yes, but then it would need to be excluded from JSON deserialization because the type is used for parsing the JSON file. 
Alternatively, `ReadLaunchProfile` can return both the `LaunchSettingsProfile` and the name in a tuple or a new `NamedLaunchSettingsProfile` type.

Reply:
I'll take a look at this. For dotnet run i did put the launch profile name on the launch profile settings. But the implementation of dotnet watch and dotnet run in this regard is quite different.

dotnet run actually had hand crafted parsing vs just using deserialisation.

Reply:
> Doesn't seem correct. If neither `--launch-profile` or `--no-launch-profile` are specified, the default (first that has Project command) profile from `launchSettings.json` is selected (if any). In that case `launchProfileName` will be null here but you'll want to set the env variable to the default profile name, no?

Give ```launchSettings.json``` as follows:

```json
{
  "profiles": {
    "First": {
      "commandName": "Project"
    },
    "Second": {
      "commandName": "Project"
    }
  }
}
```

If I invoke ```dotnet watch``` (with no arguments) then the environment variable ```DOTNET_LAUNCH_PROFILE``` will be populated with ```First```. Here is the testcase covering that:

```csharp
        [Fact]
        public async Task RunsWithDotnetLaunchProfileEnvVariableWhenNotExplicitlySpecified()
        {
            Assert.True(string.IsNullOrEmpty(Environment.GetEnvironmentVariable("DOTNET_LAUNCH_PROFILE")), "DOTNET_LAUNCH_PROFILE cannot be set already when this test is running");

            var testAsset = TestAssets.CopyTestAsset(AppName)
                .WithSource()
                .Path;

            await App.StartWatcherAsync(testAsset);
            Assert.Equal("<<<First>>>", await App.AssertOutputLineStartsWith("DOTNET_LAUNCH_PROFILE = "));
        }
```

Reply:
Hrm ... I works, but I am not sure why it works. Investigating further.

Reply:
OK I reworked this to be a bit more consistent with what I did with ```dotnet run``` where I modified ```LaunchProfileSettings``` to include the launch profile name. I think the previous approach was working/would work but it was more of a side effect of the way that we did command-line parsing rather than an explicit capture of the launch profile name that we actually use.

Reply:
> Hrm ... I works, but I am not sure why it works. Investigating further.

Note that there are two different modes `dotnet watch` can run in: 1) Hot Reload 2) legacy (any change rebuilds the app and restarts). They execute in a different way. The later uses `dotnet run` the former does not.

Reply:
Yeah. I've added logic to both code paths regardless.